umpost/um2netcdf.py

```diff
@@ -300,17 +308,8 @@ def apply_mask(c, heaviside, hcrit):
 
 
 def process(infile, outfile, args):
-    # Use mule to get the model levels to help with dimension naming
-    # mule 2020.01.1 doesn't handle pathlib Paths properly
     ff = mule.load_umfile(str(infile))
-
-    if isinstance(ff, mule.ancil.AncilFile):
-        raise NotImplementedError('Ancillary files are currently not supported')
-
-    # TODO: eventually move these calls closer to their usage
-    grid_type = get_grid_type(ff)
-    dlat, dlon = get_grid_spacing(ff)
-    z_rho, z_theta = get_z_sea_constants(ff)
+    mv = process_mule_vars(ff)
```

Comment:
Yes, this should be renamed as it's misleading. It's not processing mule vars, it's accessing them.

It could be called `extract_model_levels()`, although is that still too non-descriptive?

Reply:
I'd been thinking mostly about the variable name `mv` rather than the function, though we could rename the function too. I think the function is extracting the model level information with the `z_rho` and `z_theta` outputs, but also information about the horizontal grid with the `grid_type`, `d_lat` and `d_lon` outputs. Would something like `extract_grid_info()` or `extract_grid_metadata()` work here?

Reply:
This is somewhat fiddly as the func does 2 things: extracting z sea data & grid metadata. I'm tempted to leave it as is, given splitting the func means changing data structures & another mock (or swapping to opening/reading test data files across several tests).

How's this hybrid naming idea: `model_levels = extract_grid_model_levels()`?
The generic `MuleVars` requires a rename too. Perhaps `ModelLevels = namedtuple(...)`?

Reply:
Ah I see, happy to leave as is for the moment in that case! I think a downside of something like `extract_grid_model_levels()` is that (to me) I interpret it as extracting the "grid's model levels" rather than "grid and model levels", both of which make sense but I think only the latter would be accurate here.

Reply:
I'll reference this discussion in #27, where it can be addressed as a minor issue.